test/cli/whole-program_test.py

```diff
@@ -214,6 +216,65 @@ def test_suppress_inline_project_builddir_j(tmpdir):
     os.mkdir(build_dir)
     __test_suppress_inline_project(tmpdir, ['-j2', '--cppcheck-build-dir={}'.format(build_dir)])
 
+@pytest.mark.parametrize("jobs,builddir", ((1,False), (1,True), (2,False), (2,True)))
+def test_addon_rerun(tmpdir, jobs, builddir):
+    """Rerun analysis and ensure that misra CTU works; with and without build dir, with -j1 and -j2"""
+    args = [
+        f'-j{jobs}',
```

Comment:
We are not using format strings.

Reply:
alright :-(

Reply:
Some context: they were just introduced in Python 3.6 and we still support Python 3.5 (via Ubuntu 16.04) although we currently cannot test that in the CI.

Reply:
We currently do not have documented what Python versions are supported. Since support for Python 2.7 is removed in this development cycle I will add that information and then we can decide if we want to deprecate more afterwards.

Reply:
thanks.. for user scripts i.e. the addons I I vote to support as old python version as we can. For these test scripts I don't feel it's that important. If we can execute these test scripts in the CI with whatever python version is readily available there that is fine for me.

Reply:
> the addons I I vote to support as old python version as we can.

There is an upcoming PR which will clarify things. I want to finish up with the pre-3.x Python stuff first.